### ISSUE TITLE
Pytorchmodelcreator layer biases are not consistent with Pytorch biases

### ISSUE THREAD

**Issue by SamKG** [PrincetonUniversity/PsyNeuLink#1461]
When layer biases are initialized within PsyNeuLink during the creation of a Pytorch model in they are set to be zero (see: `pytorchmodelcreator.py:84`)

However, the standard Pytorch behaviour is to randomly initialize the biases. (Ex: 
`import torch.nn as nn
nn.Linear(1,1)` 
Is equivalent to a single neuron with linear activation, but has random weights.

Current test cases don't catch this, as they avoid using `nn.Linear` and related Pytorch layer creation methods altogether (instead opting to directly call lowlevel torch tensor operations).

We should look into ways to correct this inconsistency by allowing manual biases, and add more robust test cases that account for this.

**Comment by jdcpni** [comment]
Good catch.  Sounds like a good thing to put on the todo list for you and/or Dillon.

jdc

Sent from my iPhone

On Jan 29, 2020, at 3:57 PM, SamKG <notifications@github.com> wrote:

﻿

When layer biases are initialized within PsyNeuLink during the creation of a Pytorch model in they are set to be zero (see: pytorchmodelcreator.py:84)

However, the standard Pytorch behaviour is to randomly initialize the biases. (Ex:
import torch.nn as nn nn.Linear(1,1)
Is equivalent to a single neuron with linear activation, but has random weights.

Current test cases don't catch this, as they avoid using nn.Linear and related Pytorch layer creation methods altogether (instead opting to directly call lowlevel torch tensor operations).

We should look into ways to correct this inconsistency by allowing manual biases, and add more robust test cases that account for this.
